### ISSUE TITLE
Add CLI option for inspecting the contrib database

### ISSUE THREAD

**Issue by apiad** [autogoal/autogoal#16]
The idea is to have something like

```
$ python -m autogoal contrib list
```

This would show all the algorithms currently available under `contrib` that can be imported with the currently installed dependencies, and warnings for all the dependencies that need to be installed.

The algorithms can be obtained by calling `autogoal.contrib.find_classes()`.

```
$ python -m autogoal contrib search [--include KEYWORD] [--exclude KEYWORD] [--input INPUT_TYPE] [--output OUTPUT_TYPE]
```

This would list the algorithms that match the given keywords (e.g., `"Embedding"`) and optionally matching the input and output types.

The algorithms can be obtained by calling `autogoal.contrib.find_classes(include="...", exclude="...")`. For matching input and output types, some coding would be necessary, but I think the `Interface` class solves most of the problem.

**Comment by AdityaNikhil** [comment]
Hey @apiad , am glad I came across your repository and I feel like this will become the next big thing.
So I had some plans in my mind to develop this API like adding automated GANs architecture or automated object detection with just few lines of code. So can I work on them to integrate with this API?

Also please add **hacktoberfest-accepted** to the labels. So once if a genuine PR has been merged, you can label that PR as **hacktoberfest-accepted** . Also according to updated guidelines of hacktoberfest, **hacktoberfest-accepted** is needed to count towards contributions.

Thank you, waiting for your reply!

**Comment by apiad** [comment]
Hey, thanks for coming! Sure, anything that aligns with the AutoGOAL design is welcome! If you could describe a little bit the approach you want to use, just so we could make sure it will be compatible with what we already have, that would be great. Feel free to open a new issue and describe what you would like to do, and we can discuss a little bit the design there. 

Also thanks for the recommendation about the label. I will add it.

**Comment by AdityaNikhil** [comment]
@apiad  Sure, I'll let you know the plan and details soon.
I just came across this link, https://autogoal.github.io/dependencies/ ,
And it's written that, `AutoGOAL itself does not depend on sklearn or keras` .
So most of the architectures which I'll be creating are dependent on keras, am afraid if this affects my contribution.

**Comment by apiad** [comment]
Don't worry, that is not a problem. What this means is that AutoGOAL does not require `keras`, it is loaded as a contrib. You can add any algorithm that depends on `keras`, and the users of the library that install `keras` will be able to use it. Those that do not wish to install `keras` can use the other algorithms. That's the whole idea.
